src/load_data.cpp

```diff
@@ -0,0 +1,75 @@
+#include <iostream>
+#include <string>
+#include <filesystem>
+
+#include "../include/load_data.hpp"
+
+// namespace fs = std::filesystem;
+
+// Dataloader constructor
+Dataloader::Dataloader(std::string name, std::string path) 
+{ 
+    // Constructor sets name and path to images
+    // it also sets up the data structure that can
+    // be queried at any time to get the next image
+    dataset_name = name;
+    dataset_path = path;
+
+    // Create image vector
+    create_feature_vectors();
+}
+
+void Dataloader::create_feature_vectors() {
+    // Delete this after our two images test
+    int counter = 0;
+    bool plot = false;
+
+    cv::Ptr<cv::FeatureDetector> detector = cv::ORB::create();
+    
+    for (const auto & entry : std::filesystem::directory_iterator(dataset_path)) 
+    {
+        // --- Delete this after our two images test ---
+        if (counter > 1) {
+            break;
+        }
+        // --- Delete this after our two images test ---
+        
+        const cv::Mat image = cv::imread(entry.path(), 0); //Load as grayscale
+
+        std::vector<cv::KeyPoint> keypoints;
+        detector->detect(image, keypoints);
+
+        feature_vec.push_back( keypoints );  // add keypoints to keypoint vec
+
+        // Add results to image and save.
+        if (plot)
```

Comment:
Can you put all of this plotting logic in a separate function? Ideally we should try to keep functions no more than 20 lines or so